scripts/west_commands/runners/intel_adsp.py

```diff
@@ -122,15 +135,25 @@ def flash(self, **kwargs):
 
         # Copy the zephyr to target remote ADSP host and run
         self.run_cmd = ([f'{self.cavstool}','-s', f'{self.remote_host}', f'{send_bin_fw}'])
-        self.log_cmd = ([f'{self.cavstool}','-s', f'{self.remote_host}', '-l'])
 
-        self.logger.debug(f"cavstool({self.cavstool}), fw('{send_bin_fw})")
+        # Add the extra tool options to run/request service tool
+        if self.run_opt_args:
+            self.run_cmd = self.run_cmd + self.run_opt_args
+
         self.logger.debug(f"rcmd: {self.run_cmd}")
 
         self.check_call(self.run_cmd)
 
         # If the self.pty assigned, the output the log will
         # not output to stdout directly. That means we can
         # make the log output to the PTY.
-        if not self.pty:
+        if self.pty is not None:
```

Comment:
The comment above fails to make any difference between the target and build machine. this is very confusing.

Reply:
Thanks for pointing this out, I neglected updating this comment message. Got it fixed.